generators/upgrade/index.js

```diff
@@ -0,0 +1,180 @@
+            }.bind(this));
+        },
+
+        checkLatestVersion: function() {
+            this.log('Looking for latest ' + GENERATOR_JHIPSTER + ' version...');
+            var done = this.async();
+            shelljs.exec('npm show ' + GENERATOR_JHIPSTER + ' version', {silent:true}, function (code, stdout, stderr) {
+                this.latestVersion = stdout.replace('\n','');
+                if (this.force || semver.lt(this.currentVersion, this.latestVersion)) {
+                    this.log(chalk.green('New ' + GENERATOR_JHIPSTER + ' version found: ' + this.latestVersion));
+                } else {
+                    this.log(chalk.green('No update available'));
+                    return;
+                }
+                done();
+            }.bind(this));
+        },
+
+        detectCurrentBranch: function() {
+            var done = this.async();
+            this.git.exec('rev-parse', {q: true}, ['--abbrev-ref', 'HEAD'], function(err, msg) {
+                if (err != null) throw new Error(err);
+                this.currentBranch = msg;
+                done();
+            }.bind(this));
+        },
+
+        prepareUpgradeBranch: function() {
+            var done = this.async();
+            var createUpgradeBranch = function() {
+                this.git.exec('branch', {q: true}, [UPGRADE_BRANCH], function(err, msg) {
+                    if (err != null) throw new Error(err);
+                    this.log('Created branch ' + UPGRADE_BRANCH);
+                }.bind(this));
+            }.bind(this);
+            this.git.exec('rev-parse', {q: true}, ['--verify', UPGRADE_BRANCH], function(err, msg) {
+                if (err != null) createUpgradeBranch();
+                done();
+            }.bind(this));
+        },
+
+        checkoutUpgradeBranch: function() {
+            this.gitCheckout(UPGRADE_BRANCH);
+        }
+    },
+
+    default: {
+        insight: function () {
+            var insight = this.insight();
+            insight.trackWithEvent('generator', 'upgrade');
+        },
+
+        updateJhipster: function() {
+            this.log('Updating ' + GENERATOR_JHIPSTER + '...');
+            var done = this.async();
+            shelljs.exec('npm uninstall -g ' + GENERATOR_JHIPSTER, {silent:true}, function (code, stdout, stderr) {
+            }.bind(this));
+            shelljs.exec('npm install -g ' + GENERATOR_JHIPSTER, {silent:true}, function (code, stdout, stderr) {
+                done();
+            }.bind(this));
+            this.log(chalk.green('Updated ' + GENERATOR_JHIPSTER + ' to version ' + this.latestVersion));
+        },
+
+        cleanUp: function() {
+            var done = this.async();
+            this.git.exec('rm', {q: true}, ['-rf', '--ignore-unmatch', '*'], function(err, msg) {
+                if (err != null) throw new Error(err);
+            }.bind(this));
+            shelljs.exec('rm -rf *', {silent:true}, function (code, stdout, stderr) {
```

Comment:
Woulnt this remove .git, node_modules, .yo-rc.json etc as well, not sure about osx but it removes in windows, may be we should specify folders, files to remove